### PR TITLE
[fix] [cli] Fix Broker crashed by too much memory usage of pulsar tools

### PR DESCRIPTION
**Pull request by poorbarcode**
### Motivation
After #15868, we allow `PULSAR_MEM` & `PULSAR_GC` to be overridden in `pulsar_tool_env.sh`.

Many users set `-Xms` to `2G` or larger in `PULSAR_MEM`, this will make the tools(such as `pulsar-admin`) cost a lot of memory, and if users execute `pulsar-admin` or another tool on the machine where the Broker is deployed, the current device will not have enough memory to allocate, resulting in a broker crash.

### Modifications

When `PULSAR_MEM` is overridden  in `pulsar_tool_env.sh`, delete parameter `-Xms`

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository: 
- https://github.com/poorbarcode/pulsar/pull/85

### CODE REVIEW

File: conf/pulsar_tools_env.sh [hunk 1]
```diff
@@ -42,6 +42,18 @@
 # PULSAR_GLOBAL_ZK_CONF=
 
 # Extra options to be passed to the jvm
+# Discard parameter "-Xms" of $PULSAR_MEM, which tends to be the Broker's minimum memory, to avoid using too much
+# memory by tools.
+if [ -n "$PULSAR_MEM" ]; then
+  PULSAR_MEM_REWRITE="-Xmx128m -XX:MaxDirectMemorySize=128m"
```
Comment (coderzc) [inline conf/pulsar_tools_env.sh:48]: This shell seems to set PULSAR_MEM to `-Xmx128m -XX:MaxDirectMemorySize=128m`?

**Review by gaoran10 — APPROVED** [review]

File: conf/pulsar_tools_env.sh (hunk 1 above)
Comment (poorbarcode) [reply conf/pulsar_tools_env.sh:48]: If `PULSAR_MEM` exists, `-Xmx` and `-XX:MaxDirectMemorySize` will be overridden to `128m`, and other options in `PULSAR_MEM` will not be changed.

**Review by coderzc — APPROVED** [review]
LGTM

File: conf/pulsar_tools_env.sh (hunk 1 above)
Comment (coderzc) [reply conf/pulsar_tools_env.sh]: We can just remove the `-Xmx`, `-Xmx` and `-XX:MaxDirectMemorySize` don't need overriding to 128m

**Pull request merged**